docs/operate-and-deploy/schema-inference-with-id.md

```diff
@@ -0,0 +1,409 @@
+`title` field with default value will appear in serialized data even though inserted value can never 
+have `title` field because it's not in logical schema.
+
+Incompatible example of physical schema because of mismatched type for `pageId`:
+```sql
+{
+"schema":"{
+    \"type\":\"record\",
+    \"name\":\"PageViewNewSchema\",
+    \"namespace\":\"io.confluent.ksql.avro_schemas\",
+    \"fields\":[
+        {\"name\":\"pageId\",\"type\":\"string\",\"default\":\"id\"},
+        {\"name\":\"ts\",\"type\":\"int\",\"default\":123}
+    ]
+}"
+}
+```
+
+!!! note
+`AVRO` schema raw string in {{ site.sr }} should be single line raw string without `\n`. Above format
+is for ease of read.
+
+#### Data Serialization
+
+When schema ID is provided and schema inference is successful, data source can be successfully created.
+When writing to the data source, the physical schema inferred by the schema ID will be used to serialize
+data instead of logical schema being used in other cases. Because ksqlDB translated logical schema accepts
+`null` value but physical schema may not, serialization can fail even the inserted value is valid value
+for logical schema.
+
+For example, if physical schema is `AVRO` schema and defined as following in {{ site.sr }} with ID 1:
+
+```
+{
+"schema":"{
+    \"type\":\"record\",
+    \"name\":\"PageViewValueSchema\",
+    \"namespace\":\"io.confluent.ksql.avro_schemas\",
+    \"fields\":[
+        {\"name\":\"page_name\",\"type\":\"string\"},
+        {\"name\":\"ts\",\"type\":\"int\"}
+    ]
+}"
+}
+```
+
+!!! note
+`AVRO` schema raw string in {{ site.sr }} should be single line raw string without `\n`. Above format
+is for ease of read.
+
+and `CREATE` statement is defined as following:
+
+```sql
+CREATE STREAM pageviews (
+    pageId INT KEY
+  ) WITH (
+    KAFKA_TOPIC='pageviews-avro-topic',
+    KEY_FORMAT='KAFKA',
+    VALUE_FORMAT='AVRO',
+    VALUE_SCHEMA_ID=1,
+    PARTITIONS=1
+  );
+```
+
+then the schema for `pageviews` will be inferred as:
+
+```
+ksql> describe pageviews;
+
+Name                 : PAGEVIEWS
+ Field     | Type
+------------------------------------
+ PAGEID    | INTEGER          (key)
+ page_name | VARCHAR(STRING)
+ ts        | INTEGER
+------------------------------------
+```
+
+When you insert values to `pageviews` with `null` value, it will return error:
+
+```
+ksql> insert into pageviews values (1, null, null);
+Failed to insert values into 'PAGEVIEWS'. Could not serialize value: [ null | null ]. Error serializing message to topic: pageviews-avro-topic1. Invalid value: null used for required field: "page_name", schema type: STRING
+```
+
+This is because `page_name` and `ts` are required fields without default values in original physical schema.
+
+!!! important
+ksqlDB doesn't check physical schema referred to by schema ID that its fields are all optional or with 
+a default value to make sure `null` can be serialized for flexibility of physical schema. Users are supposed
```

Comment:
This is actually an interesting point: does it imply that if a field is required with default value, but the query tries to insert a `NULL`, the `NULL` would be replaced with the default? Or would the whole record be dropped? -- I think the later might actually be better (not sure)? \cc @colinhicks

Reply:
> for flexibility of physical schema.

Not sure what this means?

Reply:
inserting null for require field will fail actually.